### ISSUE TITLE
NextJS + React Library with styled-components breaks.

### ISSUE THREAD

**Issue by Epenance** [nrwl/nx#4573]
## Current Behavior
When using styled-components in a NextJS app locally it works flawless, but if you generate a React library with styled-components and import components from it in your app, it will break on refresh randomly or each time you edit your theme config.

## Expected Behavior

I'd expect imported components from my libary to work the same way as the ones in the app itself.

## Steps to Reproduce

Run `nx serve core` in the following repo: https://github.com/Epenance/nx-styled-components/

Try to refresh a few times, if it doesn't break try to go into `/apps/core/pages/_app.tsx` and change the [bgColor ](https://github.com/Epenance/nx-styled-components/blob/a6675fb2750cc0ac77eb5f382f0a161ccbcae147/apps/core/pages/_app.tsx#L24) in the theme settings. 

You should immediatly see that the background color in the header has stopped working and the console should throw an error about className not matching between server and client.

### Failure Logs

![image](https://user-images.githubusercontent.com/6709028/105216825-5c118f80-5b53-11eb-86a3-b2ed349a41a8.png)

### Environment

```
Node : 12.18.3
  OS   : win32 x64
  npm  : 6.14.6

  nx : Not Found
  @nrwl/angular : Not Found
  @nrwl/cli : 11.1.5
  @nrwl/cypress : 11.1.5
  @nrwl/devkit : 11.1.5
  @nrwl/eslint-plugin-nx : 11.1.5
  @nrwl/express : Not Found
  @nrwl/jest : 11.1.5
  @nrwl/linter : 11.1.5
  @nrwl/nest : Not Found
  @nrwl/next : 11.1.5
  @nrwl/node : Not Found
  @nrwl/react : 11.1.5
  @nrwl/schematics : Not Found
  @nrwl/tao : 11.1.5
  @nrwl/web : 11.1.5
  @nrwl/workspace : 11.1.5
  typescript : 4.0.5

```

**Comment by atligudl** [comment]
Same issue as https://github.com/nrwl/nx/issues/2573

No solutions so far 🤔

**Comment by LuckeeDev** [comment]
Had the same issue and after investigating for a bit it turned out it was a problem in my Babel configuration.
By default, after bootstrapping an nx workspace with nextjs and styled-components, the `.babelrc` looks like this:
```json
"plugins": [
    ["styled-components", { "pure": true, "ssr": true }]
]
```
But it should look like this to make it work properly:
```json
"plugins": [
    ["babel-plugin-styled-components", { "pure": true, "ssr": true }]
]
```

Seems like there's an error in the way the nx cli bootstraps the project. Thoughts on this?

**Comment by jaysoo** [comment]
It shouldn't matter if the plugin is prefixed with `babel-plugin-` or not since babel will resolve them correctly with or without.

It looks like the issue is that babelrc in the libs are not being picked up since next.js is not using the root `babel.config.json`.

See my comment here for a workaround for now. We'll fix it in the next version. 

https://github.com/nrwl/nx/issues/4611#issuecomment-788990096

**Comment by LuckeeDev** [comment]
Thanks @jaysoo

**Comment by kirjai** [comment]
Fixed by #4944
